api/clients/v2/payloadretrieval/relay_payload_retriever.go

```diff
@@ -189,6 +197,13 @@ func (pr *RelayPayloadRetriever) retrieveBlobWithTimeout(
 	}
 
 	blob, err := coretypes.DeserializeBlob(blobBytes, blobLengthSymbols)
+	if errors.Is(err, coretypes.ErrBlobLengthSymbolsNotPowerOf2) {
+		panic(fmt.Errorf(
+			"retrieveBlobWithTimeout: blobLengthSymbols=%d is not power of 2: "+
+				"this is a major broken invariant, that should have been checked by the validators, "+
+				"and the caller (GetEncodedPayload) should already have checked this invariant "+
+				"and returned a MaliciousOperatorsError", blobLengthSymbols))
+	}
```

Comment:
nit - why put panic vs log error and return 500 via proxy? I guess since blob length is preimage input to blob key calculation used for retrieval lookups then it's guaranteed to be tamper resistant by the client with the first invariant check. iiuc you've modified retrievals like:
```
1. checkPow2(blobLengthSymbols)
2. fetch blob
3. checkPow2(blobLengthSymbols) when deserializing
```

also in the worst case if this is called it would just crash the proxy's routed handler go routine and not the core daemon server process. in that circumstance what would the client see / how would it interpet? or is the idea that this code would never ever be trigged which is why we embed panic to explicitly highlight the extreme irregularity of this conditional being triggered?

Reply:
panic is equivalent to returning 500 + logging error (actually stack trace)! See https://pkg.go.dev/net/http#Handler
<img width="989" height="111" alt="image" src="https://github.com/user-attachments/assets/4f814394-713f-49ca-9eba-e6a5114d1b47" />

However you did make me realize that if the invariant is ever broken.. probably still better to return the same MaliciousOperatorErr to user anyways... especially in case we ever decide to handle those specially in some middleware.